ls1intum/artemis-android
# `Feature`: Add Unordered and Ordered List Options to Formats

### Problem Description
Our mobile markdown editor lacked support for creating ordered or unordered lists, a feature already available in the web editor. Mobile users were unable to format lists conveniently, requiring them to manually type prefixes like -  or 1. , which disrupted the ease of composing structured messages.

### Changes
 
**Ordered List:** Adds logic to insert a 1.  prefix (or increment an existing number, e.g. 2.  -> 3. ) when the user selects “Ordered List.”

**Unordered List:** Adds logic to insert a -  prefix.

**Formatting Dropdown:** Introduces a single “List” icon in the toolbar. Tapping it reveals a dropdown to choose between unordered/ordered lists.

### Steps for testing
1.	Open any conversation
2.	Tap the “List” icon. Verify a dropdown appears with “Unordered List” and “Ordered List.”
3.	Choose “Unordered List” to insert or apply -  at the cursor.
4.	Choose “Ordered List” to insert or apply 1.  (or increment an existing number).
6.	Check auto-continuation: If a line starts with -  or N. , pressing Enter automatically inserts another bullet or the next number.

### Screenshots


https://github.com/user-attachments/assets/8a8cc416-2607-4588-93d5-1fb6d637204b

<img width="250" alt="Screenshot 2024-12-22 at 21 31 05" src="https://github.com/user-attachments/assets/89516545-325e-442b-84ae-747e305388c1" />
